Gallery-Server/src/main/java/edu/yu/cs/gallery/GalleryResource.java

```diff
@@ -149,22 +151,39 @@ private long registerWithHub(URL url) throws UnknownHostException {
         return response;
     }
 
-    //Hub talks to the gallery
+    //Hub talks to the gallery and sets the map of the IDs to the URLs.
     @POST 
     @Path("/servers")
     public Response updateIPs(Map<Long, URL> as) {
         utility.allServers = as;
-        if (utility.allServers.equals(utility.allServers)) {
+        if (utility.allServers.equals(as)) {
             return Response.status(Status.OK).build();
         }
         return Response.status(INTERNAL_SERVER_ERROR).build();
     }
     
+    // Hub talks to the gallery and sets the leaderID.
+    @POST 
+    @Path("/leader")
+    public Response updateLeader(Long leaderID) {
+        utility.leaderID = leaderID;
+        if (utility.leaderID.equals(leaderID)) {
```

Comment:
would this ever not be true?

Reply:
Yeah. We were just trying to make it consistent with the updateIPs method above. Just changed it so it only has the first line of code and returns void. Should we make the updateIPs method also only have that one line of code that sets it equal to the map

Reply:
yeah, I cant see a great reason for the check in either place.

Reply:
Got it. Deleted from both of them